tensorflow/c/eager/dlpack.cc

```diff
@@ -0,0 +1,335 @@
+      status->status = tensorflow::errors::InvalidArgument(
+          "TF_QINT8 is not supported by dlpack");
+      break;
+    case TF_DataType::TF_QUINT8:
+      status->status = tensorflow::errors::InvalidArgument(
+          "TF_QUINT8 is not supported by dlpack");
+      break;
+    case TF_DataType::TF_QINT32:
+      status->status = tensorflow::errors::InvalidArgument(
+          "TF_QINT32 is not supported by dlpack");
+      break;
+    case TF_DataType::TF_BFLOAT16:
+      dtype.code = DLDataTypeCode::kDLBfloat;
+      break;
+    case TF_DataType::TF_QINT16:
+      status->status = tensorflow::errors::InvalidArgument(
+          "TF_QINT16 is not supported by dlpack");
+      break;
+    case TF_DataType::TF_QUINT16:
+      status->status = tensorflow::errors::InvalidArgument(
+          "TF_QUINT16 is not supported by dlpack");
+      break;
+    case TF_DataType::TF_COMPLEX128:
+      status->status = tensorflow::errors::InvalidArgument(
+          "TF_COMPLEX128 is not supported by dlpack");
+      break;
+    case TF_DataType::TF_HALF:
+      dtype.code = DLDataTypeCode::kDLFloat;
+      break;
+    case TF_DataType::TF_RESOURCE:
+      status->status = tensorflow::errors::InvalidArgument(
+          "TF_RESOURCE is not supported by dlpack");
+      break;
+    case TF_DataType::TF_VARIANT:
+      status->status = tensorflow::errors::InvalidArgument(
+          "TF_VARIANT is not supported by dlpack");
+      break;
+    case TF_DataType::TF_UINT32:
+      dtype.code = DLDataTypeCode::kDLUInt;
+      break;
+    case TF_DataType::TF_UINT64:
+      dtype.code = DLDataTypeCode::kDLUInt;
+      break;
+    default:
+      status->status = tensorflow::errors::InvalidArgument(
+          "Unsupported TF_DataType is not supported by dlpack");
+      break;
+  }
+  return dtype;
+}
+
+DLContext  GetDLContext(TFE_TensorHandle* h, TF_Status* status) {
+  DLContext ctx;
+  const char* device_name = h->handle->DeviceName(&status->status);
+  DeviceNameUtils::ParsedName parsed_name;
+  tensorflow::DeviceNameUtils::ParseFullName(absl::string_view(device_name),
+                                             &parsed_name);
+  std::string device_type = parsed_name.type;
+  int device_id = -1;
+  if (parsed_name.has_id) {
+    device_id = parsed_name.id;
+  }  // Question? device_id?=-1
+
+  ctx.device_id = device_id;
+  if (device_type == "CPU") {
+    ctx.device_type = DLDeviceType::kDLCPU;
+  } else if (device_type == "GPU") {
+    ctx.device_type = DLDeviceType::kDLGPU;
+  } else {
+    status->status = tensorflow::errors::InvalidArgument(
+        "Unsupported Device Type for DLPack");
+  }
+
+  return ctx;
+}
+
+DLManagedTensor* TFEHandleToTFDLManagedTensorCtx(TFE_TensorHandle* h,
+                                        TF_Status* status) {
+  const Tensor* tensor = GetTensorFromHandle(h, status);
+  TF_DataType data_type = static_cast<TF_DataType>(tensor->dtype());
+  TFDLManagedTensorCtx* tfDLMTensor(new TFDLManagedTensorCtx);
```

Comment:
Why not

```
auto* tfDLMTensor = new TFDLManagedTensorCtx;
```

Also Google style is `tf_dlm_tensor_ctx` or `tf_dl_managed_tensor_ctx` if that's not too verbose.